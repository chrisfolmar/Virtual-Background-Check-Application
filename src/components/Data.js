import React, { Component } from 'react'

/**
 * A component to represent a data node
 */
 class Data extends Component {

	/**
     * render
     * @return {ReactElement} markup
     */
     render(){
     	const header = this.props.header,
     	icon = this.props.icon,
     	data = this.props.data;

     	return(
     		<div className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3">
	     		<div className="features-icons-icon d-flex">
	     			<i className={icon +" m-auto text-primary"}></i>
	     		</div>
	     		<h5>{header}</h5>
	     		<p className="mb-0">{data}</p>
     		</div>
     		);
     	}
     }

     export default Data;