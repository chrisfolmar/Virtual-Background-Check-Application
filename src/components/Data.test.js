import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import Data from './Data';

const header = "Hostname",
icon = "icon-list",
data = "Test Data";

describe('Data', () => {

	it('renders fully without crashing', () => {
		const component = mount(
			<Data 
				header={header} 
				icon={icon} 
				data={data}
			/>
			);
		component.unmount();
	});

	it('full snapshot rendering', () => {
		const component = mount(<Data />);
		expect(mountToJson(component)).toMatchSnapshot();
		component.unmount();
	});

	it('renders with propogated data', () => {
		const component = mount(
			<Data 
				header={header} 
				icon={icon} 
				data={data}
			/>
			);
		const populatedDataComponentElement = <p className="mb-0">{data}</p>;
		expect(component.contains(populatedDataComponentElement)).toEqual(true);
	});

});