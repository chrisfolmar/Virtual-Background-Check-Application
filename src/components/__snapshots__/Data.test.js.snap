// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data full snapshot rendering 1`] = `
<Data>
  <div
    className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
  >
    <div
      className="features-icons-icon d-flex"
    >
      <i
        className="undefined m-auto text-primary"
      />
    </div>
    <h5 />
    <p
      className="mb-0"
    />
  </div>
</Data>
`;
